core/server/src/main/java/alluxio/master/file/meta/InodeDirectory.java

```diff
@@ -212,6 +214,27 @@ public static InodeDirectory fromJournalEntry(InodeDirectoryEntry entry) {
     return inode;
   }
 
+  /**
+   * Creates an {@link InodeDirectory}.
+   *
+   * @param id id of this inode
+   * @param parentId id of the parent of this inode
+   * @param name name of this inode
+   * @param directoryOptions options to create this directory
+   * @return the {@link InodeDirectory} representation
+   */
+  public static InodeDirectory create(long id, long parentId, String name,
```

Comment:
should we change the constructor to private to enforce that all objects are created using this factory?